sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs

```diff
@@ -82,6 +82,8 @@ void AddCustomerPolicies(HttpPipelinePosition position)
                         }
                     }
                 }
+
+                policies.RemoveAll(static policy => policy == null);
```

Comment:
Instead of calling removeAll we can check for null inside the loop and not add a null element. Or check for null in ClientOptions.AddPolicy